nvflare/private/fed/server/job_cmds.py

```diff
@@ -196,6 +196,20 @@ def _start_app_on_server(self, conn: Connection, job_id: str) -> bool:
 
     def _start_app_on_clients(self, conn: Connection, job_id: str) -> bool:
         engine = conn.app_ctx
+        client_names = conn.get_prop(self.TARGET_CLIENT_NAMES, None)
+        participants = engine.run_processes.get(job_id, {}).get(RunProcessKey.PARTICIPANTS, [])
+        wrong_clients = []
+        for client in client_names:
+            for _, p in participants.items():
```

Comment:
Addressed.

Reply:
Not resolved since you are still iterate thru the "participants" dict, which could change size by another thread.
I think, you want:

`participant_names = list(participants.keys())
  
for name in participant_names:
    if client == name:
        ....`

Reply:
No, RunProcessKey.PARTICIPANTS in the run_process won't be changed in another thread. Only the engine.run_processes could potentially be changed. Once run_process object exist, it's safe to loop through the participants. There's already protection check on the "if not run_process:"